### DISCUSSION TITLE
Waiting for a volume to be created either by the external provisioner 'rook-ceph.rbd.csi.ceph.com'

### DISCUSSION THREAD

**Discussion by BlauerFasan** [rook/rook#14861] [Q&A]
Good Morining Rook Comunity

I'm new to Rook & Ceph, I like the idea very much. But i ran in to some Problem i have no clue how to get to the rootcause of it. I really would appriciate some troubleshooting help. 
I'm stuck with this problem now for longer than a week and tried different things incluing several cluster redeployments, and a lot of changes to the configuration.

I found here in the questions as well as in the docu somethings but somehow it seems to me that it does not help me with a fix.
i followed this links and helps to get it solved: 

- https://github.com/rook/rook/discussions/7614
- https://rook.github.io/docs/rook/v1.5/ceph-common-issues.html#pvcs-stay-in-pending-state
- https://rook.github.io/docs/rook/v1.5/ceph-csi-troubleshooting.html#csi-common-issues

I found some other issues but they seem often about other topics like the bucket of the fs storage.

In General the Cluster of Ceph looks Healty to me:
![grafik](https://github.com/user-attachments/assets/7ba925f6-a962-4811-8888-966ad5c24239)
i tried a new setting wich led to recreation of the mon pods.

Here the Problem i facing it stays in this state forever
![grafik](https://github.com/user-attachments/assets/12a6bf51-ece3-4e4c-b48a-05bb9298493b)
What does me confuse is that the two Volumes created with the volsync config do get Bound. 
This is the closer look at it:
![grafik](https://github.com/user-attachments/assets/fe14481c-450f-4cfe-bb27-b3f6cf31f6e7)

The OSD Prepare pods Completes sucsessfully: 
![grafik](https://github.com/user-attachments/assets/70c79217-cbfa-4697-bf4d-df4fee599491)
in the Examples they show often that there is 0/2 but i have 0/1 is that a Problem or what is causes the difference? 

The CSI Provisioner are running as well: 
![grafik](https://github.com/user-attachments/assets/c946cca3-01ac-45e9-93c3-6ff86f7e738f)

I added the logs of the CSI- Provisioner, the Operatior as well as the cluster configuration in as attach
[CSI Provisioner Log.txt](https://github.com/user-attachments/files/17393193/CSI.Provisioner.Log.txt)
[Operator Log.txt](https://github.com/user-attachments/files/17393249/Operator.Log.txt)

Thank you for your help in advance, i'm looking forward to your response.

**Comment by travisn** [comment]
The `influxdb` pvc is not getting bound because of this message seen in the csi provisioner log: 
```
I1016 09:17:12.653374       1 event.go:377] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"database", 
Name:"influxdb", UID:"13613d44-83e2-4845-925d-ffb74c1d6573", APIVersion:"v1", ResourceVersion:"386623", FieldPath:""}): 
type: 'Normal' reason: 'Provisioning' Assuming an external populator will provision the volume
```

@Rakshith-R What external provisioner would that be referring to? The PVC seems to just be coming from the rook-ceph-block storage class.

Note that the latest csi troubleshooting guide is [here](https://rook.io/docs/rook/latest/Troubleshooting/ceph-csi-common-issues/). The links you shared are very old documentation.

**Reply by Rakshith-R** [reply]
It should be cephcsi driver but it referring to `external populator will provision the volume` .
I don't know how VolSync works but we need full PVC yaml to figure out.
It's probably something on VolSync side.

**Reply by BlauerFasan** [reply]
Thank you both for your time to look in to this,

As @travisn mentioned i had an old troubleshooting guid i followed, so i did it with the new one you added. I got stuck at the connections to the mon endpoint. 
'# Test the network connection to the mon endpoint' 
Which i can not get the desired result, I get 'curl: (1) Received HTTP/0.9 when not allowed'
This let me dig the last night in to cilium which is my cluster network provider. Let me deploy hubble and run different conncetivity tests. 
![grafik](https://github.com/user-attachments/assets/354538c7-b501-4254-9845-02ea73dffc65)
I think it forwards the desired traffic to the mon pod as mentioned.

@Rakshith-R As you requested i added the PVC yaml, for the pending PVC as well as the other two shown in the Picture above which get bound.
[influxdb-pvc.txt](https://github.com/user-attachments/files/17421431/influxdb-pvc.txt)
[influxdb-ReplicationDestination.txt](https://github.com/user-attachments/files/17421432/influxdb-ReplicationDestination.txt)
[influxdb-replication-source.txt](https://github.com/user-attachments/files/17421433/influxdb-replication-source.txt)

**Comment by Rakshith-R** [comment]
It needs to be looked at from volsync side.
Volsync is actually "populating" the volume 
https://kubernetes.io/blog/2021/08/30/volume-populators-redesigned/

```
I1016 09:17:12.653374       1 event.go:377] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"database", 
Name:"influxdb", UID:"13613d44-83e2-4845-925d-ffb74c1d6573", APIVersion:"v1", ResourceVersion:"386623", FieldPath:""}): 
type: 'Normal' reason: 'Provisioning' Assuming an external populator will provision the volume
```

```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    volume.beta.kubernetes.io/storage-provisioner: rook-ceph.rbd.csi.ceph.com
    volume.kubernetes.io/storage-provisioner: rook-ceph.rbd.csi.ceph.com
  creationTimestamp: "2024-10-17T05:28:01Z"
  finalizers:
  - kubernetes.io/pvc-protection
  - snapshot.storage.kubernetes.io/pvc-as-source-protection
  labels:
    app.kubernetes.io/name: influxdb
    kustomize.toolkit.fluxcd.io/name: influxdb
    kustomize.toolkit.fluxcd.io/namespace: flux-system
  name: influxdb
  namespace: database
  resourceVersion: "11857"
  uid: 36044e0e-c241-473a-a87b-8409f696050d
spec:
  accessModes:
  - ReadWriteOnce
  dataSource:
    apiGroup: volsync.backube
    kind: ReplicationDestination
    name: influxdb-bootstrap
  dataSourceRef:
    apiGroup: volsync.backube
    kind: ReplicationDestination
    name: influxdb-bootstrap
  resources:
    requests:
      storage: 10Gi
  storageClassName: ceph-block
  volumeMode: Filesystem
```
